.github/workflows/ci-dgo-tests.yml

```diff
@@ -0,0 +1,59 @@
+name: ci-dgo-tests
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+    branches:
+      - main
+  schedule:
+    - cron: "0 4 * * *" # daily at 4:00
+jobs:
+  dgo-tests:
+    runs-on: [self-hosted, x64] # needed to use docker commands
+    steps:
+      - name: Checkout Dgraph repo
+        uses: actions/checkout@v3
+        with:
+          path: dgraph 
+          repository: dgraph-io/dgraph
+          ref: main
+      - name: Checkout Dgo repo
+        uses: actions/checkout@v3
+        with:
+          path: dgo
+          repository: dgraph-io/dgo
+      - name: Move into Dgraph repo
+        run: cd dgraph 
+      - name: Get Go Version
+        run: |
+          #!/bin/bash
+          GOVERSION=$({ [ -f .go-version ] && cat .go-version; })
+          echo "GOVERSION=$GOVERSION" >> $GITHUB_ENV
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GOVERSION }}
+      - name: Make Linux Build and Docker Image
+        run: make docker-image
+      - name: Move into dgo repo
+        run: cd ../dgo
+      - name: Clean Up Environment
+        run: |
+          #!/bin/bash
+          # clean go test cache
+          go clean -testcache
+      - name: Run dgo client tests
+        run: |
+          #!/bin/bash
+          docker compose -f t/docker-compose.yml up -d
+          echo "Waiting for cluster to be healthy..."
+          sleep 20
```

Comment:
Ugly, I know - but we can improve this later.

Reply:
can you explore this https://gist.github.com/rgl/f90ff293d56dbb0a1e0f7e7e89a81f42